src/ethereum_test_tools/tests/test_filler_expect.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test suite for `ethereum_test` module.
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork, Istanbul, London, Merge, Shanghai
+from evm_transition_tool import GethTransitionTool
+
+from ..common import Account, Environment, Transaction
+from ..common.types import Storage
+from ..filling import fill_test
+from ..spec import BaseTestConfig, StateTest
+from ..vm.opcode import Opcodes as Op
+
+
+@pytest.mark.parametrize(
+    "fork,enable_hive",
+    [
+        (Istanbul, False),
+        (London, False),
+        (Merge, True),
+        (Shanghai, True),
+    ],
+)
+def test_post_expect_section(fork: Fork, enable_hive: bool):
+    """
+    Test `ethereum_test.filler.fill_fixtures` with `StateTest` and post state verification.
+    """
+    env = Environment()
+
+    pre = {
+        0x1000000000000000000000000000000000000000: Account(code=Op.SSTORE(1, 1)),
+        "0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b": Account(balance=1000000000000000000000),
+    }
+
+    tx = Transaction(
+        ty=0x0,
+        chain_id=0x0,
+        nonce=0,
+        to="0x1000000000000000000000000000000000000000",
+        gas_limit=100000000,
+        gas_price=10,
+        protected=False,
+    )
+
+    post = {
+        "0x1000000000000000000000000000000000000000": Account(storage={"0x01": "0x02"}),
+    }
+
+    state_test = StateTest(
+        env=env,
+        pre=pre,
+        post=post,
+        txs=[tx],
+        tag="my_chain_id_test",
+        base_test_config=BaseTestConfig(enable_hive=enable_hive),
+    )
+
+    t8n = GethTransitionTool()
+
+    wasError = False
+    try:
+        fixture = {
+            f"000/my_chain_id_test/{fork}": fill_test(
+                t8n=t8n,
+                test_spec=state_test,
+                fork=fork,
+                spec=None,
+            ),
+        }
+        fixture.clear()
+    except Storage.KeyValueMismatch as e:
+        wasError = True
+        assert e.want == 2
+        assert e.got == 1
+        assert e.key == 1
+        assert e.address == "0x1000000000000000000000000000000000000000"
+    assert wasError
```

Comment:
pytest has a helper [`pytest.raises`](https://docs.pytest.org/en/latest/how-to/assert.html#assertions-about-expected-exceptions) to check that correct exceptions get raised which provides more informative info than manually checking:
```suggestion
    e_info: pytest.ExceptionInfo
    with pytest.raises(Storage.KeyValueMismatch) as e_info:
        fixture = {
            f"000/my_chain_id_test/{fork}": fill_test(
                t8n=t8n,
                test_spec=state_test,
                fork=fork,
                spec=None,
            ),
        }
    assert e_info.value.want == 2
    assert e_info.value.got == 1
    assert e_info.value.key == 1
    assert e_info.value.address == "0x1000000000000000000000000000000000000000"
    assert "incorrect value in address" in str(e_info.value)
```